distributed/executor.py

```diff
@@ -193,6 +193,22 @@ def release(self):
             self._cleared = True
             self.executor._dec_ref(tokey(self.key))
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state.update({'event': self.event.is_set(), 'status': self.status})
+        del state['executor']
+        return state
```

Comment:
If we can get away with it I recommend only sending along the Key, and then recreating everything else on the other side.  State like the event or status are not guaranteed to be valid on the other end.